amethyst_renderer/src/resources.rs

```diff
@@ -83,8 +83,8 @@ impl ScreenDimensions {
     /// Creates a new screen dimensions object with the given width and height.
     pub fn new(w: u32, h: u32, hidpi: f64) -> Self {
         ScreenDimensions {
-            w: w as f64,
-            h: h as f64,
+            w: f64::from(w),
```

Comment:
hmm not sure if this one is much better

Reply:
Yeah it doesn't look as good, but apparently doing it this way is considered best practice. From the [clippy lint docs](https://rust-lang.github.io/rust-clippy/master/index.html#cast_lossless):

> Rust’s as keyword will perform many kinds of conversions, including silently lossy conversions. Conversion functions such as i32::from will only perform lossless conversions. Using the conversion functions prevents conversions from turning into silent lossy conversions if the types of the input expressions ever change, and make it easier for people reading the code to know that the conversion is lossless.